modules/navigation/godot_navigation_server.cpp

```diff
@@ -808,8 +814,117 @@ void GodotNavigationServer::process(real_t p_delta_time) {
 			active_maps_update_id[i] = new_map_update_id;
 		}
 	}
+
+#ifdef MODULE_NAVIGATION_ENABLED
+	_process_path_queries();
+#endif // MODULE_NAVIGATION_ENABLED
+}
+
+/////////////////////////////////////
+#ifdef MODULE_NAVIGATION_ENABLED
+void GodotNavigationServer::_query_add_processqueue(const gd::PathQueryParameters &p_query_parameters, const ObjectID &p_result_object_id) const {
+	gd::PathQueryTask *pathquery_task = new gd::PathQueryTask();
+
+	pathquery_task->query_parameters = p_query_parameters;
+	pathquery_task->result_object_id = p_result_object_id;
+
+	pathquery_process_queue.push_back(pathquery_task);
+};
+
+gd::PathQueryResult GodotNavigationServer::_query_path(const gd::PathQueryParameters &p_parameters) const {
+	gd::PathQueryResult r_query_result;
+
+	const NavMap *map = map_owner.get_or_null(p_parameters.map);
+	ERR_FAIL_COND_V(map == nullptr, r_query_result);
+
+	// run the pathfinding
+
+	if (p_parameters.pathfinding_algorithm == gd::PathfindingAlgorithm::PATHFINDING_ALGORITHM_ASTAR) {
+		r_query_result.path = map->get_path(p_parameters.start_position, p_parameters.target_position, p_parameters.optimize, p_parameters.navigation_layers);
+	} else {
+		return r_query_result;
+	}
+
+	// add path postprocessing
+
+	// add calculated path stats
+
+	if (r_query_result.path.size() > 1 && p_parameters.calculate_path_length) {
+		_calculate_path_length(p_parameters, r_query_result);
+	}
+
+	return r_query_result;
+}
+
+void GodotNavigationServer::_calculate_path_length(const gd::PathQueryParameters &p_parameters, gd::PathQueryResult &p_query_result) const {
+	float _sum = 0.0;
```

Comment:
Should these be `real_t`?

Reply:
Not sure if it should but I guess it could.

Reply:
It's going to be calculated as 64 bit anyway if `Vector3` is 64 bit, so storing as 32 bit would be throwing away data in this case. I think it was stored as `real_t` elsewhere (in the query result or something)?

Reply:
real_t it is then